### DISCUSSION TITLE
Supabase Auth - make password optional on SignUp

### DISCUSSION THREAD

**Discussion by enricoschaaf** [supabase/supabase#460] [Feature Requests] [answered]
This would allow people to use Magic Link Auth only if they like to, currently you have to provide some random string as a password for that.

**Comment by kiwicopple** [comment]
Hey @enricoschaaf, we already have magic links! 

https://supabase.io/docs/client/auth-signin#sign-in-with-magic-link

**Reply by enricoschaaf** [reply]
I am aware, but a password is still required for signups, I would like to only use magic link auth, currently I just provide a random password on signup but it doesn't fell very clean that way since the password will never be used.

**Reply by kiwicopple** [reply]
That shouldn't be the case .. you should be able to get in without a password.

See here: https://supabase-nextjs-todo-list.vercel.app/
Code here: https://github.com/supabase/supabase/tree/master/examples/nextjs-todo-list

**Comment by awalias** [comment, marked as answer]
hey @enricoschaaf this functionality is already available in the js lib:

```javascript
// if no user exists will sign up and send magic link
let { user, error } = await supabase.auth.signIn({
  email: 'someone@email.com'
})
```

and raw http:

```bash
curl -X POST 'https://jxslqjbqjxjveydildym.supabase.net/auth/v1/magiclink' \
-H "apikey: SUPABASE_KEY" \
-H "Content-Type: application/json" \
-d '{ 
  "email": "someone@email.com" 
}'
```

if no user exists they will sign the user up and send a sign up confirmation email, you can also skip confirmation emails by disabling them from the dashboard:

<img width="676" alt="image" src="https://user-images.githubusercontent.com/458736/103459150-e1e4ac80-4d04-11eb-95a5-215f04e948a3.png">

This will confirm the user and send them a magic link immediately (although the confirmation email link will also return an access token when clicked by the user, so the functionality is similar)

let me know if any issues!

**Comment by enricoschaaf** [comment]
Ah, this makes so much sense, I wasn't aware of the fact that if you sign in a user without an account the account will automatically created. Neat! Thank you both very much for this clarification. ☺️

**Reply by eifo** [reply]
I have a particular scenario where I need to be able to get the auth.user back when signing in with magic links, I still need the user to verify the email (don't want to disable email confirmations) but I would like to do some work with the generated user beforehand, how can it be done?

currently the user is not returned...

let { user, error } = await supabase.auth.signIn({
  email: 'someone@email.com'
})

**Reply by pichichi91** [reply]
I just checked it in the source (https://github.com/supabase/gotrue-js/blob/master/src/GoTrueClient.ts), you never have the data while using magicLink: 

```js
if (email && !password) {        
    const { error } = await this.api.sendMagicLinkEmail(email, {          
        redirectTo: options.redirectTo,        
    })        
    return { data: null, user: null, session: null, error }      
}

```

If you want to do something with a user, you could  do something after the login (before rendering the page) - I think there's the first time you'd have access to the user for the first time in case am I not mistaken

**Reply by eifo** [reply]
Thanks for confirming from goTrue docs, yeah, so there is no way to have user returned, I can think of some workarounds but would have been great to be able to get when the unvalidated user is generated.
